src/main/kotlin/Actions/GetJsonResponseAction.kt

```diff
@@ -0,0 +1,21 @@
+package Actions
+
+import StructuredDataGenerators.JsonGenerator
+import contenttype.HttpContentTypes
+import response.ResponseBuilder
+
+class GetJsonResponseAction(jsonGenerator: JsonGenerator) : Action {
+    private val headers = mapOf("Content-Type" to (HttpContentTypes.JSON.type + HttpContentTypes.JSON.parameter))
+    private val responseBody = jsonGenerator.generateJson(mapOf("key1" to "value1", "key2" to "value2"))
+    private lateinit var requestBody: String
+
+    override fun act(responseBuilder: ResponseBuilder): String {
+        responseBuilder.setHeaders(headers)
+        responseBuilder.setBody(responseBody)
+        return responseBuilder.build()
+    }
+
+    override fun setBody(body: String) {
+        requestBody = body
+    }
```

Comment:
Is this function every called?

Reply:
It is called in the `routeRequest` method. It is used to pass any body message from the client to the `Action`

Reply:
Ahh totally didn't read it closely. I think it would be nice if it was called `setRequestBody`. I was thinking this was the `responseBody`. Or passing this into the constructor (if possible) might make it easier to understand.

Either way, it's a big refactor, so don't do it in this PR. If you decide you want to make the change, I would open a new PR.